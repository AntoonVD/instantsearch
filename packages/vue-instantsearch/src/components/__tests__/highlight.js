import Vue from 'vue';
import {
  Highlight,
  HIGHLIGHT_PRE_TAG,
  HIGHLIGHT_POST_TAG,
} from 'vue-instantsearch';

test('renders proper HTML', () => {
  const result = {
    _highlightResult: {
      attr: {
        value: `con${HIGHLIGHT_PRE_TAG}ten${HIGHLIGHT_POST_TAG}t`,
      },
    },
  };

  const vm = new Vue({
    template: '<highlight attributeName="attr" :result="result">',
    render(h) {
      return h('highlight', {
        props: {
          attributeName: 'attr',
          result: result,
        },
      });
    },
    components: {
      Highlight,
    },
  }).$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('accepts custom highlighting tag', () => {
  const result = {
    _highlightResult: {
      attr: {
        value: `con${HIGHLIGHT_PRE_TAG}ten${HIGHLIGHT_POST_TAG}t`,
      },
    },
  };

  const vm = new Vue({
    template: '<highlight attributeName="attr" :result="result">',
    render(h) {
      return h('highlight', {
        props: {
          attributeName: 'attr',
          result: result,
          tagName: 'em',
        },
      });
    },
    components: {
      Highlight,
    },
  }).$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('protects against XSS', () => {
  const result = {
    _highlightResult: {
      attr: {
        value: "<script>alert('test');</script>",
      },
    },
  };

  const vm = new Vue({
    template: '<highlight attributeName="attr" :result="result">',
    render(h) {
      return h('highlight', {
        props: {
          attributeName: 'attr',
          result: result,
        },
      });
    },
    components: {
      Highlight,
    },
  }).$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('allows unsafe output', () => {
  const result = {
    _highlightResult: {
      attr: {
        value: '<strong>un-escaped output</strong>',
      },
    },
  };

  const vm = new Vue({
    template: '<highlight attributeName="attr" :result="result">',
    render(h) {
      return h('highlight', {
        props: {
          attributeName: 'attr',
          result: result,
          escapeHtml: false,
        },
      });
    },
    components: {
      Highlight,
    },
  }).$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('should render an empty string if attribute is not highlighted', () => {
  const result = {
    _highlightResult: {},
  };

  const vm = new Vue({
    template: '<highlight attributeName="attr" :result="result">',
    render(h) {
      return h('highlight', {
        props: {
          attributeName: 'attr',
          result: result,
        },
      });
    },
    components: {
      Highlight,
    },
  }).$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});
