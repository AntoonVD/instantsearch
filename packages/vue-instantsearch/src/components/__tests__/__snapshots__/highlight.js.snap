// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts custom highlighting tag 1`] = `

<span class="ais-highlight">
  con
  <em>
    ten
  </em>
  t
</span>

`;

exports[`allows unsafe output 1`] = `

<span class="ais-highlight">
  <strong>
    un-escaped output
  </strong>
</span>

`;

exports[`protects against XSS 1`] = `

<span class="ais-highlight">
  &lt;script&gt;alert('test');&lt;/script&gt;
</span>

`;

exports[`renders proper HTML 1`] = `

<span class="ais-highlight">
  con
  <mark>
    ten
  </mark>
  t
</span>

`;

exports[`should render an empty string if attribute is not highlighted 1`] = `

<span class="ais-highlight">
</span>

`;
